plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -701,7 +864,44 @@ int process_and_send(struct flb_cloudwatch *ctx, struct cw_flush *buf,
             continue;
         }
 
-        ret = add_event(ctx, buf, stream, &map, &tms);
+        if (strncmp(input_plugin, "cpu", 3) == 0 
+            || strncmp(input_plugin, "mem", 3) == 0) {
+            /* Added for EMF support: Construct a list */
+            struct mk_list flb_intermediate_metrics;
+            mk_list_init(&flb_intermediate_metrics);
+
+            kv = map.via.map.ptr;
+
+            /* 
+             * Iterate through the record map, extract intermediate metric data, 
+             * and add to the list.
+             */
+            for (i = 0; i < map_size; i++) {
+                metric = flb_malloc(sizeof(struct flb_intermediate_metric));
+                if (!metric) {
+                    flb_errno();
+                    return;
+                }
```

Comment:
for all error cases, before returning, you need to clean up and free memory

in this function, the error label takes care of that; so use `goto error` like the other code does

Reply:
Updated.